### PR TITLE
Update SensitiveDataLogger for dual environment support

### PR DESCRIPTION
**Pull request by marcominerva**
Modified `EnsureDevelopmentEnvironment `to check `DOTNET_ENVIRONMENT` if `ASPNETCORE_ENVIRONMENT` is not set, and updated exception message. Updated **SensitiveDataLoggerTests.cs** to distinguish between environment variables. Added new test methods for different environments. Renamed existing test methods for clarity. Updated `Dispose` method to clear both environment variables.

## Motivation and Context (Why the change? What's the scenario?)
See https://github.com/microsoft/kernel-memory/issues/851

### CODE REVIEW

File: service/tests/Abstractions.UnitTests/Diagnostics/SensitiveDataLoggerTests.cs [hunk 1]
```diff
@@ -71,9 +106,50 @@ public void ItCanBeDisabledInAnyEnvironment(string environment)
...
+
+        // Assert
+        Assert.False(SensitiveDataLogger.Enabled);
+    }
+
+    [Theory]
+    [Trait("Category", "UnitTest")]
+    [InlineData("development", "staging")]
+    [InlineData("Development", "Staging")]
+    public void AspNetCoreEnvironmentHasThePrecedence(string aspNetCoreEnvironment, string dotNetEnvironment)
```
Comment (dluc) [inline service/tests/Abstractions.UnitTests/Diagnostics/SensitiveDataLoggerTests.cs:135]: this doesn't seem to test the precedence, does it?

Comment (dluc) [reply service/tests/Abstractions.UnitTests/Diagnostics/SensitiveDataLoggerTests.cs:135]: I believe this should be a couple of tests

##1

        Environment.SetEnvironmentVariable(AspNetCoreEnvironmentVariableName, "development");
        Environment.SetEnvironmentVariable(DotNetEnvironmentVariableName, "production");
        SensitiveDataLogger.Enabled = true;
         => can be enabled



##2

        Environment.SetEnvironmentVariable(AspNetCoreEnvironmentVariableName, "production");
        Environment.SetEnvironmentVariable(DotNetEnvironmentVariableName, "development");
        Assert.Throws<InvalidOperationException>(() => SensitiveDataLogger.Enabled = true);
         => throws

Comment (marcominerva) [reply service/tests/Abstractions.UnitTests/Diagnostics/SensitiveDataLoggerTests.cs:135]: > this doesn't seem to test the precedence, does it?

It tests that the `ASPNETCORE_ENVIRONMENT`, that is set to **Development**, takes the precendence over `DOTNET_ENVIRONMENT`, that is set to **Staging**. But of course we can make two separate tests.

File: service/tests/Abstractions.UnitTests/Diagnostics/SensitiveDataLoggerTests.cs [hunk 2]
```diff
@@ -71,9 +106,50 @@ public void ItCanBeDisabledInAnyEnvironment(string environment)
...
+        Assert.False(SensitiveDataLogger.Enabled);
+        Environment.SetEnvironmentVariable(AspNetCoreEnvironmentVariableName, aspNetCoreEnvironment);
+        Environment.SetEnvironmentVariable(DotNetEnvironmentVariableName, dotNetEnvironment);
+
+        // Act
+        SensitiveDataLogger.Enabled = true;
+
+        // Assert
+        Assert.True(SensitiveDataLogger.Enabled);
+    }
```
Comment (dluc) [inline service/tests/Abstractions.UnitTests/Diagnostics/SensitiveDataLoggerTests.cs:147]:
```suggestion
    [Theory]
    [Trait("Category", "UnitTest")]
    [InlineData("development", "staging", false)]
    [InlineData("Development", "Staging", false)]
    [InlineData("development", "production", false)]
    [InlineData("Development", "Production", false)]
    [InlineData("staging", "development", true)]
    [InlineData("Staging", "Development", true)]
    [InlineData("production", "development", true)]
    [InlineData("Production", "Development", true)]
    public void AspNetCoreEnvironmentHasThePrecedence(string aspNetCoreEnvironment, string dotNetEnvironment, bool fail)
    {
        // Arrange
        Assert.False(SensitiveDataLogger.Enabled);
        Environment.SetEnvironmentVariable(AspNetCoreEnvironmentVariableName, aspNetCoreEnvironment);
        Environment.SetEnvironmentVariable(DotNetEnvironmentVariableName, dotNetEnvironment);

        if (fail)
        {
            Assert.Throws<InvalidOperationException>(() => SensitiveDataLogger.Enabled = true);
        }
        else
        {
            // Act
            SensitiveDataLogger.Enabled = true;

            // Assert
            Assert.True(SensitiveDataLogger.Enabled);
        }
    }
```

**Review by dluc — APPROVED** [review]

**Pull request merged**